Feature/authorization

naturalnumber opened the pull request:
Pull request to bring dev up to the current functionality

naturalnumber committed:
Serializer fixes

MatthewBowlan committed:
UserSurveyHistory and SampleDataGenerator
Added a BASIC view, serializer, and url for UserSurveyHistory
Added Sample Data Generator Script

MatthewBowlan committed:
- Profile endpoint - serializer, view, url
- Noticed that the router urls seam off

MatthewBowlan committed:
- added default permission classes
- Available Surveys - serializers, views, urls

MatthewBowlan committed:
- Indicator basic endpoint - not functioning

naturalnumber committed:
Analysis system core

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
Serializer fix

MatthewBowlan committed:
- added default permission classes
- Available Surveys - serializers, views, urls

MatthewBowlan committed:
- added python file like load_surveys.py to load profiles for the 5 example users

naturalnumber committed:
Namable glitch patch

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
git_ignore fix

naturalnumber committed:
typo fix

naturalnumber committed:
defaults fix

naturalnumber committed:
defaults fix

naturalnumber committed:
typo fix

naturalnumber committed:
remove debug code

naturalnumber committed:
Available surveys format change

naturalnumber committed:
fix refactor error

naturalnumber committed:
fix annotation serialization glitch

naturalnumber committed:
Several minor fixes, submission parsing correctly

naturalnumber committed:
typo fix

MatthewBowlan committed:
- 2 surveys added to load_surveys.py

naturalnumber committed:
parsing fixes

naturalnumber committed:
debug change

kareemabulez committed:
First survey for the Lung (LINQ)

naturalnumber committed:
validation typo

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

lmsch committed:
fix(models): potential fix?

MatthewBowlan committed:
- random password generator added to sample_data_generation.sh
- also includes an included function to normalize emails

MatthewBowlan committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
typo fix in boolean static method

naturalnumber committed:
Finished attaching analysis system

naturalnumber committed:
Fixed glitch in requirements file script

kareemabulez committed:
added two remaining surveys

naturalnumber committed:
Fix survey glitch

naturalnumber committed:
Profile Retrieval endpoint change

naturalnumber committed:
Merge branches 'Development' and 'feature/authorization' of https://github.com/naturalnumber/Revitalize into feature/authorization

 Conflicts:
	Revitalize/models.py
	Revitalize/serializers.py

naturalnumber committed:
Merged Dev

lmsch commented on backend_api/settings.py:
```diff
@@ -128,9 +136,11 @@
-CORS_ORIGIN_WHITELIST = [
-        'http://localhost:3000',
-]
+CORS_ORIGIN_ALLOW_ALL = True
```
Probably should revert this back to whitelist.

naturalnumber replied:
Oops

naturalnumber committed:
Apostrophe fix

naturalnumber committed:
CORS settings fix

naturalnumber committed:
Analysis skeleton, validation redo, translation fix

naturalnumber committed:
typo fix

MatthewBowlan committed:
- survey scoring added to analysis_helper.py

naturalnumber committed:
gad7, phq9 scoring

MatthewBowlan committed:
- fixed load_surveys.py error

naturalnumber committed:
url fix

naturalnumber committed:
alt clean

lmsch committed:
fix(models): change apos

naturalnumber committed:
disable new addresses for demo

lmsch committed:
fix(surveys): add utf-8 char

lmsch committed:
fix(survey): remove invalid utf-16

naturalnumber committed:
Survey History, empty response error return fix

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
multi error glitch patch

naturalnumber committed:
Validation and response recursion clean-up.
Analysis system glitch fixed.

naturalnumber committed:
Error type glitch

naturalnumber committed:
turned of debug

MatthewBowlan committed:
- Profile Pictures Working

naturalnumber committed:
Error fix

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
Restricted survey history

naturalnumber committed:
boolean fix

naturalnumber committed:
dummy lab value end point

naturalnumber committed:
address system live

naturalnumber committed:
address system live

MatthewBowlan committed:
Adjusted time field in UserSurveyHistorySerializer

naturalnumber committed:
processing flag glitch fix

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
Change in error return format

naturalnumber committed:
Analysis system upgrades

naturalnumber committed:
Indicator endpoint upgrades

naturalnumber committed:
Labvalue endpoint upgrades

naturalnumber committed:
Timezone fix

MatthewBowlan committed:
Added an admin page for lab tech - currently shows how models can only be registered for one site.

naturalnumber committed:
graphing endpoint changes

MatthewBowlan committed:
Changed Lab Tech Admin pages to be user friendly and restricting access to many functions
Changed Admin Site to include all models while allowing customizing where needed
- Any admin user can access both the Lab Tech and the Admin page: need to limit labtechs to just the lab tech site

naturalnumber committed:
requested endpoint changes

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
endpoints cut down

MatthewBowlan committed:
commented code in admins site to stop lab tech from accessing full admin site... need to use abstract user instead of base user so that a 'is_labtech' boolean can be added

naturalnumber committed:
endpoints minor change

naturalnumber committed:
typo and documentation

naturalnumber committed:
view class merge

naturalnumber committed:
view class de-merge

MatthewBowlan committed:
admin and labtechs different access levels... breaks database

naturalnumber committed:
copy-past error fix, endpoint trimming

naturalnumber committed:
Merge remote-tracking branch 'origin/feature/authorization' into feature/authorization

naturalnumber committed:
time zone typo

naturalnumber committed:
method replacement in user model

naturalnumber committed:
profile serialier update

naturalnumber committed:
various cleaning

naturalnumber committed:
indicator lookup logging

naturalnumber committed:
missed if branch
